### PR TITLE
spec: 2.0.0 send transaction

### PR DESCRIPTION
**Pull request by alaibe**
#68

### CODE REVIEW

File: content/raw/wallet/2.0.0_send_transaction.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+## Motivation
+
+One of the most common use cases for a wallet is to send tokens to another account. A user typically expects to know how much tokens they are sending are worth in some currency and also expect a wide range of functionality such as ENS addresses, address validation using EIP-55, and the ability to choose easily common addresses, saved or recently used address to send to. The user also expects to be able to understand the transactions fees involved and choose the priority for transaction.
+
+## Definitions
+
+### Best route
+
+We define the best route as the less costly route amongst all the possible route
```
Comment (Khushboo-dev-cpp) [inline content/raw/wallet/2.0.0_send_transaction.md:26]: wouldn't this be a time to less costly trade off?

File: content/raw/wallet/2.0.0_send_transaction.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+Swap is the action of creating one or more transactions in order to bridge a token from one chain to another chain
+
+### Adddress field
+
+The address field can be filled with multiple option:
+1. an ethereum address
+2. an ENS name
+3. a Contact display name
+
+Plus it can be prefixed by the destination chains of the transaction separated by `:`
```
Comment (Khushboo-dev-cpp) [inline content/raw/wallet/2.0.0_send_transaction.md:47]: will we write down the various destinations chains supported in the feature specs?

**Review by Khushboo-dev-cpp — COMMENTED** [review]
Awesome! some small comments from me?
Are we also going to mention the scope of chains supported by us?
Initially it will be 3 right?

File: content/raw/wallet/2.0.0_send_transaction.md (hunk 2 above)
Comment (alaibe) [reply content/raw/wallet/2.0.0_send_transaction.md:47]: Yes I need to add specs for chain supported and bridge supported

File: content/raw/wallet/2.0.0_send_transaction.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+9. (optional) User selects Advance routing and is able to inspect the routing in details
+10. (optional) User selects Custom routing and is able to make the route of his choice based on the possible route returned by the system
+11. System displays Priority section with "Optimal" pre selected
+12.1 (optional) alternative: User selects a priority
+12.2 (optional) alternative: User selects priority for each transaction
+13. System updates the estimated time
+10. System validates if origin account has enough to pay for gas
+11. User presses Send
+12. System notifies user transaction is ongoing
+13. System notifies user when transaction is complete
```
[the remark this answers is not in the document]
Comment (alaibe) [reply content/raw/wallet/2.0.0_send_transaction.md:97]: I need to check with @carlfairclough @John-44 about the UI when there is multiple transaction involved, how the UI shouldd look like

File: content/raw/wallet/2.0.0_send_transaction.md (hunk 1 above)
Comment (alaibe) [reply content/raw/wallet/2.0.0_send_transaction.md:26]: I was thinking as an option here to either choose between time or cost

Comment (Khushboo-dev-cpp) [reply content/raw/wallet/2.0.0_send_transaction.md:26]: optimal I thought would be best tradeoff between time and cost. like perhaps not the cheapest but something with better time than the cheapest option? 

I don't see design for the user to make choice between cost and time yet, right?

**Comment by alaibe** [comment]
@Khushboo-dev-cpp just added first chain/bridge supported this will probably evolve but this is a start

File: content/raw/wallet/2.0.0_send_transaction.md (hunk 1 above)
Comment (alaibe) [reply content/raw/wallet/2.0.0_send_transaction.md:26]: I thought about that too, choosing something having the best trade off, this means we have to bias our algorithm. How do u compare time and cost? is 10 gwei  for 5 minutes better than 20 gwei for 2 minutes 30 seconds?

As u can see we would have to apply a multiplicator between time and cost. TBH I have no idea what it should be I even believe it is different depending on who u are. So I kinda avoid to make a decision by having this option

WDYT?

> I don't see design for the user to make choice between cost and time yet, right?

No design yet indeed, I was thinking it is only a toggle into wallet settings

Comment (alaibe) [reply content/raw/wallet/2.0.0_send_transaction.md]: Resolved: best route in terms of cost in a reasonable time

File: content/raw/wallet/2.0.0_send_transaction.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (alaibe) [reply content/raw/wallet/2.0.0_send_transaction.md]: Resolved, use history view, design is pending

**Pull request closed without merging**